### ISSUE TITLE
Message queue trigger should handle connection lost situation (NATS)

### ISSUE THREAD

**Issue by life1347** [fission/fission#950]
A user on slack reports the message queue trigger no longer trigger function when PING requests to nats streaming server failed.

```
time="2018-10-22T17:15:06Z" level=warning msg="Failed to publish message to topic new-product: stan: connection lost due to PING failure"
time="2018-10-22T17:15:06Z" level=warning msg="Failed to publish message to topic new-kpi: stan: connection lost due to PING failure"
```

After some tracing, the go library(go-nats-streaming) after `0.4.0` added ability to send PING requests to nats streaming server to detect connection lost. The PR https://github.com/nats-io/go-nats-streaming/pull/183 allows us to add callback function to handle such situation when all PINGs failed.

**Comment by gameiro** [comment]
Hi @vishal-biyani  what is the status of this issue ? 
I reported the problem in slack and we are still having a lot of problems with this error , our message queue trigger is failing and the functions are not triggering.  Is there a workarround for this problem ? 

The only thing the helps when this ping error appears is to restart the mqt trigger pod, is it possible to add some health check command to restart the pod in case of error message ?

**Comment by vishal-biyani** [comment]
Hi, @gameiro Thanks for the details, we will try to get this fixed in 1.1 or 1.2

**Comment by life1347** [comment]
Hi @gameiro 
Since it's kind of annoy problem, I try to address this problem by add connection lost handle in PR https://github.com/fission/fission/pull/1125 which will restart mqtrigger automatically. I think we are able to ship this fix in milestone 1.1, which is close to the due day. For more detail, please see PR comment.
